src/services/nationalCenters.ts

```diff
@@ -1,59 +1,155 @@
 import { auditLogger } from '../logger';
+import SCOPES from '../middleware/scopeConstants';
 import db from '../models';
 
-const { NationalCenter } = db;
+const {
+  NationalCenter, User, Permission, NationalCenterUser,
+} = db;
+
+const { READ_WRITE_TRAINING_REPORTS } = SCOPES;
 
 interface NationalCenterType {
   name: string;
   id: number;
-  mapsTo?: number;
+  users?: [{
+    id: number;
+    name: string;
+  }],
+}
+
+interface NationalCenterUserType {
+  name: string;
+  id: number;
 }
+
 interface NCModel extends NationalCenterType {
   destroy: (args: { individualHooks: boolean }) => Promise<void>;
   save: () => Promise<void>;
-  update: (args: { mapsTo: number }, options: { individualHooks: boolean }) => Promise<void>;
+  update: (args: {
+    id: number,
+    name: string,
+  }, options: { individualHooks: boolean }) => Promise<void>;
 }
 
 export async function findAll(): Promise<NationalCenterType[]> {
   return NationalCenter.findAll({
-    where: {
-      mapsTo: null,
-    },
-    attributes: ['id', 'name', 'mapsTo'],
+    attributes: ['id', 'name'],
+    include: [{
+      // National Center Users.
+      model: User,
+      as: 'users',
+      attributes: ['id', 'name'],
+    }],
     order: [['name', 'ASC']],
   });
 }
 
+export async function findAllNationalCenterUsers(): Promise<NationalCenterUserType[]> {
```

Comment:
Feels to me like a user service, getUserByPermission(READ_WRITE_TRAINING_REPORTS), if we want to be able to reuse it

Reply:
Are you saying we want to write a generic function for this? I hesitate to do so because I feel they might change this to be more TR specific down the road.

Reply:
you think we'll need to do something more than lookup a user by permission or set of permissions? what makes you think that?

Reply:
Just seems a bit loose currently ie not region specific.